### PR TITLE
[JENKINS-29308] Update project configuration

### PR DESCRIPTION
**Pull request by recena**
https://issues.jenkins-ci.org/browse/JENKINS-29308

- [x] Configure Findbugs Maven Plugin
- [x] Update the version of Maven Release Plugin
- [x] Configure License information
- [x] Review the coding style (partial)

### CODE REVIEW

**Comment by amuniz** [comment]
It's really difficult to review using `w=1` because that view does not allow to add comments.

File: src/main/java/hudson/plugins/logparser/LogParserAction.java [hunk 1]
```diff
@@ -1,232 +1,230 @@
...
+
+    public String getUrlName() {
+        return urlName;
+    }
+
+    public static String getUrlNameStat() {
+        return urlName;
+    }
+
+    public Run<?, ?> getOwner() {
```
Comment (amuniz) [inline src/main/java/hudson/plugins/logparser/LogParserAction.java]: `@WithBridgeMethods` should be used here (something like [this](https://github.com/jenkinsci/analysis-core-plugin/blob/42c50ab8f5cca9da0d83519a03afd7fe2a46de57/src/main/java/hudson/plugins/analysis/views/ConsoleDetail.java#L114)).

File: src/main/java/hudson/plugins/logparser/LogParserBuildStep.java [hunk 2]
```diff
@@ -1,40 +1,37 @@
...
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.tasks.BuildStep;
+import hudson.tasks.BuildStepMonitor;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+public class LogParserBuildStep implements BuildStep {
```
Comment (amuniz) [inline src/main/java/hudson/plugins/logparser/LogParserBuildStep.java]: Should not it extend `SimpleBuildStep`?

File: src/main/java/hudson/plugins/logparser/LogParserPublisher.java [hunk 3]
```diff
@@ -1,232 +1,284 @@
...
-        else
-            return null;
-    }
-
-}
+package hudson.plugins.logparser;
+
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.*;
```
Comment (amuniz) [inline src/main/java/hudson/plugins/logparser/LogParserPublisher.java]: Not recommended.

File: src/main/java/hudson/plugins/logparser/LogParserPublisher.java [hunk 4]
```diff
@@ -1,232 +1,284 @@
...
+    }
+
+    @Deprecated
+    public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
+            throws InterruptedException, IOException {
+
+        if (build.getResult() == Result.SUCCESS) {
+            this.perform(build, build.getWorkspace(), launcher, listener);
+        } else {
+            listener.error("Skipping publisher since build result is " + build.getResult());
```
Comment (amuniz) [inline src/main/java/hudson/plugins/logparser/LogParserPublisher.java]: Why there is still logic in a deprecated method? Unless I'm missing something, I think it should pass directly to the new implementation.

**Comment by amuniz** [comment]
Reviewed as good as I could given that there is a lot of formatting changes (even shown with `w=1`) and it's really difficult to see where "real" logic changes have been made.

It seems that the build is failing, so delaying my bee until it passes.

**Comment by recena** [comment]
@amuniz I'm going to create a new PR focused on "workflow compatibilty".

**Comment by recena** [comment]
@amuniz Finaly, I've decided to add a new PR https://github.com/jenkinsci/log-parser-plugin/pull/10 /cc @jglick

File: src/main/java/hudson/plugins/logparser/CompiledPatterns.java [hunk 5]
```diff
@@ -1,33 +1,37 @@
...
+
+public class CompiledPatterns {
+
+    private String errorMsg;
+    private Pattern[] compiledPatterns;
+
+    public CompiledPatterns() {
+        this.errorMsg = null;
+        this.compiledPatterns = null;
+    }
```
Comment (tfennelly) [inline src/main/java/hudson/plugins/logparser/CompiledPatterns.java]: Why make this change?

**Comment by tfennelly** [comment]
Hey Manuel ... it's quite hard to see what's new/changed here Vs just reformatting of existing code (which seems like the biggest part of the diff). It's probably more trouble than it's worth at this stage, but wouldn't it be easier to do the reformatting as a separate PR?

**Comment by tfennelly** [comment]
And in general ... I'm not sure what the general motivation is here. General cleanup?

File: src/main/java/hudson/plugins/logparser/LogParserBuildStep.java (hunk 2 above)
Comment (vaskrist) [inline src/main/java/hudson/plugins/logparser/LogParserBuildStep.java]: I have not seen this class referenced anywhere and it does not seem to do anything at all, does it?

**Comment by KostyaSha** [comment]
Is there any reason for creating JENKINS-29308 ? 
Why did you remove PMD plugin?

**Comment by oleg-nenashev** [comment]
:bug: The code is not readable. The codestyle modifications hide other changes, which actually require the review. Codestyle should be determined by the plugin maintainers, so I see no reason for changing it now. It will break pull requests submitted by others.

**Comment by recena** [comment]
@oleg-nenashev I'm waiting for the maintainer.

**Comment by jglick** [comment]
Agreed, this is unreviewable as is, mostly reformatting?

**Comment by KostyaSha** [comment]
@glick well, what is see:
- [x] Remove Maven PMD plugin

**Comment by oleg-nenashev** [comment]
Yeah, a removal of a static analysis tool smells bad. What's the justification?

**Comment by KostyaSha** [comment]
So your personal refactoring made other PRs unmergeable?

**Comment by recena** [comment]
@KostyaSha I know it but I sent an email to the PR owners in order to help them. And yes, it is my personal refactoring because I don't like to work with source code without a basic rules (styles).

**Pull request merged**